src/python/pants/engine/isolated_process.py

```diff
@@ -132,6 +131,19 @@ class FallibleExecuteProcessResult:
     stderr: bytes
     exit_code: int
     output_directory_digest: Digest
+
+
+@dataclass(frozen=True)
+class FallibleExecuteProcessResultWithPlatform:
```

Comment:
While this is quite wordy, this name fits well with conventions we've been using in V2 for when you want to have some extra information attached to a normal type, e.g. `AddressWithOrigin` and `HydratedTargetWithOrigin`.

Reply:
Yeah it is wordy, but I don't have a better name in mind.